packages/react/src/wallet/infrastructure/SignerFactory.ts

```diff
@@ -95,7 +95,7 @@ export class SignerFactory implements ISignerFactory {
     try {
       if (signer.provider && signer.provider instanceof JsonRpcProvider) {
         await signer.provider.send('wallet_switchEthereumChain', [
-          { chainId: chainConfig.chainId },
+          { chainId: chainConfig.hexadecimalChainId },
```

Comment:
Thank you @jarrodwatts for fixing this.

I think we can simplify this by converting the number to hex string on the fly via:
```suggestion
          { chainId: utils.hexValue(chainConfig.chainId) },
```

This is a very low frequency operation that the extra overhead is negligible. The ethers `utils` is already used in this file.

In this case we can revert the config objects changes.